### DISCUSSION TITLE
Improving Response Times for Reverse Geocoding on a Self-hosted Nominatim Server

### DISCUSSION THREAD

**Discussion by dlubom** [osm-search/Nominatim#3401] [Q&A]
Hello Nominatim Community,

We have recently installed Nominatim with the complete global dataset on our server, and I have been conducting some benchmark tests comparing the response times for reverse geocoding between our server and the official Nominatim server.

Here are the results for 100 randomly generated lat/lon coordinates, with requests sent sequentially:

Our Server:

- Total execution time: 19.56 seconds
- Average time per request: 0.20 seconds
- Minimum time: 0.02 seconds
- Maximum time: 0.90 seconds

Official Nominatim Server:

- Total execution time: 93.66 seconds
- Average time per request: 0.94 seconds
- Minimum time: 0.15 seconds
- Maximum time: 1.79 seconds

Given these results, I am exploring ways to further improve the response times on our server. Our current setup includes a fast hybrid disk and 128 GB of RAM, with 10 CPU cores. I am considering several upgrades, such as switching to an SSD, increasing RAM, or simply optimizing default settings.

Would any of these changes significantly enhance our server's performance, especially given our plan to handle millions of daily unique requests? Any suggestions or insights on the potential impact of these upgrades would be greatly appreciated.

Thank you for your help!

**Comment by lonvia** [comment]
Use NVME SSDs if you can. It makes a huge difference. More RAM will also help.

For specs of our servers, browse https://hardware.openstreetmap.org/ (look out for "Nominatim").
